### DISCUSSION TITLE
Why is buffer size private

### DISCUSSION THREAD

**Discussion by Stock84-dev** [gfx-rs/wgpu#2096] [Q&A] [answered]
I see that `total_size` is stored inside `MapContext`. Is there a reason why users cannot access buffer size with something like `buffer.size()`.

**Comment by kvark** [comment, marked as answer]
Just consistency. That's a part of the descriptor that users passed in, so they know it already. If we start exposing *some* of the information we store, then we'll need to figure out where to stop. For examples, do bind groups need to carry information about all the bindings? And similar. So today we aren't exposing any of that, consistently.
